File: src/beginner_tutorials/src/Esteban_talker.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
#include <sstream>

int main(int argc, char **argv)
{

  ros::init(argc, argv, "Esteban_talker");

  ros::NodeHandle n;

  ros::Publisher pub = n.advertise<geometry_msgs::Twist>("cmd_vel", 1000);

  ros::Rate loop_rate(1);

  int count = 0;
  float velx = 0.0, velz = 0.0;
  while (ros::ok()){
    geometry_msgs::Twist msg;
    if(count<5){
      velx = 0.5;
      velz = 0.0;
    }else if (count>5 && count<10){
      velx = 0.0;
      velz = 1.0;
    }else if (count>10 && count<15){
      velx = -0.5;
      velz = 0.0;
    }else if (count>15 && count<20){
      velx = 0.0;
      velz = -1.0;
    }else if(count == 20){
      count = 0;
    }

    msg.linear.x = velx;
    msg.angular.z = velz;

    pub.publish(msg);

    loop_rate.sleep();
    ++count;
  }
  
  return 0;
}

File: src/beginner_tutorials/scripts/Esteban_talker.py
#!/usr/bin/env python
import rospy
from geometry_msgs.msg import Twist

def talker():
    pub = rospy.Publisher('cmd_vel', Twist, queue_size=10)
    rospy.init_node('Esteban_talker_py', anonymous=True)
    rate = rospy.Rate(1) # 1hz
    msg = Twist()
    count = 0

    while not rospy.is_shutdown():
        if count<5:
            msg.linear.x=0.5
            msg.angular.z=0.0
        elif count>5 and count<10:
            msg.linear.x=0.0
            msg.angular.z=1.0
        elif count>10 and count <15:
            msg.linear.x=-0.5
            msg.angular.z=0.0
        elif count>15 and count<20:
            msg.linear.x=0.0
            msg.angular.z=-1.0
        elif count == 20:
            count=0
        pub.publish(msg)
        rate.sleep()
        count+=1

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass
